crates/esthri/src/compression.rs

```diff
@@ -25,17 +25,13 @@ pub const ESTHRI_METADATA_COMPRESS_KEY: &str = "esthri_compress_version";
 
 pub async fn compress_to_tempfile(path: &Path) -> Result<(TempFile, u64)> {
     debug!("compressing: {}", path.display());
-    let dir = match path.parent() {
```

Comment:
similarly, this should probably be 

```rust
    let dir = match (Config::global().temp_dir_path(), path.parent()) {
        (Some(dir), _) => dir,
        (None, Some(parent)) => parent.to_owned(),
        _ => std::env::current_dir()?,
    };
```

Reply:
i.e. prefer writing to ESTHRI_TEMP_DIR if its set, then prefer creating the temp file next to the original, finally default to current_dir (which realistically shouldn't happen because the parent directory should always be set)